index.d.ts

```diff
@@ -0,0 +1,14 @@
+// Type definitions for pino-caller 3.1
+// Project: https://github.com/pinojs/pino-caller#readme
+// Definitions by: Austin Ziegler <https://github.com/halostatue>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import Pino = require('pino');
+
+export = Caller;
```

Comment:
Could you please avoid using this style of export which relies on interop being enabled, and instead do `the infamous triplet`?

See https://github.com/pinojs/sonic-boom/blob/master/types/index.d.ts and https://github.com/pinojs/sonic-boom/blob/2150933ffb2108fd9804e99d94035f706fce00db/index.js#L400

Reply:
What is the "infamous triplet"?

Reply:
See examples linked. TS- wise this is a combination of a default and named export, index.js-wise it is a specific way to set both default and named export for single-export project.

Reply:
I read the linked examples, and I still don’t exactly see what the infamous triplet is, and my searches are failing me on this. Your comment here is the first time that I’ve ever seen that term used _anywhere_.

Reply:
Not sure if I ever read about it outside of fastify ecosystem, sorry for the confusion.
You can see https://github.com/fastify/fastify-swagger/pull/282/files for a (hopefully) clear example of what that is.

Reply:
So it would be:

```
export function Caller(logger: Pino.Logger, options?: Options): Pino.Logger;
export { Caller }
```

?

What should the exported name be, in this case? `PinoCaller` instead of `Caller`? I’m looking at https://github.com/pinojs/pino-caller/blob/master/index.js#L27 and seeing that the function is called `traceCaller` and is only exported as the default.

1. Should this change the export of the library in `index.js` so that there’s a pair of exports?
2. Should I just export the default in the types file until someone wants to do the pair of exports with an “official” name.

I prefer option 2, as this change is turning out to be much larger than I had planned to do.

I’m not trying to be difficult; I based the pino-caller types on https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/pino/index.d.ts in DefinitelyTyped and opened a ticket here a courtesy. The original changes took less than an hour; I’ve already spent three times that on these changes and it would be easier for me to simply remove pino-caller as I shift to typescript than to continue with this PR.

I’m more than happy to leave the branch open, and the PR should be set so that pino-caller contributors can edit this, but I don’t have time to work on this anymore. Sorry.

Reply:
@halostatue Thank you so much for all your efforts! They are dearly appreciated. No worries, if you are OK with me taking over the PR, I can add the finishing touches myself.